war/src/main/js/components/search-bar/index.js

```diff
@@ -0,0 +1,117 @@
+import { createElementFromHtml } from "@/util/dom";
+import makeKeyboardNavigable from "@/util/keyboard";
+
+const SELECTED_CLASS = "jenkins-search__results-item--selected";
+
+function init() {
+  const searchBarInputs = document.querySelectorAll(".jenkins-search__input");
+
+  Array.from(searchBarInputs)
+    .filter((searchBar) => searchBar.suggestions)
+    .forEach((searchBar) => {
+      const searchWrapper = searchBar.parentElement.parentElement;
+      const searchResultsContainer = createElementFromHtml(
+        `<div class="jenkins-search__results-container"></div>`
+      );
+      searchWrapper.appendChild(searchResultsContainer);
+      const searchResults = createElementFromHtml(
+        `<div class="jenkins-search__results"></div>`
+      );
+      searchResultsContainer.appendChild(searchResults);
+
+      searchBar.addEventListener("input", () => {
+        const query = searchBar.value.toLowerCase();
+
+        // Hide the suggestions if the search query is empty
+        if (query.length === 0) {
+          hideResultsContainer();
+          return;
+        }
+
+        showResultsContainer();
+
+        function appendResults(container, results) {
+          results.forEach((item, index) => {
+            container.appendChild(
+              createElementFromHtml(
+                `<a class="${index === 0 ? SELECTED_CLASS : ""}" href="${
+                  item.url
+                }"><div>${item.icon}</div>${item.label}</a>`
```

Comment:
Could you escape the `item.label` variable? It's not dangerous here, since the values are hardcoded, but this code could be reused for other pages with search capabilities, where the labels might be dangerous and thus introduce an XSS.

Reply:
Good idea! Done.